src/index.ts

```diff
@@ -1,3 +1,5 @@
 export { CheckpointTrie } from './checkpointTrie'
 export { SecureTrie } from './secure'
 export { Trie as BaseTrie } from './baseTrie'
+import WalkStrategy from './util/walkStrategy'
+export { WalkStrategy }
```

Comment:
why add WalkStrategy to the exported API, do you think others will use it? if so we should probably add some notes to the readme and changelog.

Reply:
Yes, client will use this in BeamSync.

Reply:
It also allows consumers to setup their own fancy ways to setup walkTrie. You can implement DFS/BFS walking of the trie using it.

Reply:
oh i see it in the OP:
> Also explicitly exposes walkTrie and lookupNode for our consumers. WalkStrategy is now also exposed. This has to do with Beam Sync in the client.

Reply:
Have added this to changelog and also included it in the docs.

Reply:
ok great thanks for explaining, that makes a lot of sense. so if someone customizes the WalkController, is there a way to feed it back into creating a trie with it? or how would that process look.

Reply:
The `WalkController` only resolves if all the promises of the task executor are finished (the queue is empty). These are all async functions. So, for instance, it is (almost) directly possible to create a beam-sync way to download the trie AND feed back into the trie: just call `walkTrie` from the root. Each time a node is found (can be `null`) then the `onNode` method is called. If the provided node is `null`, then asynchronously fetch the trie node, put it into the DB, and call `onlyBranchIndex` again for the corresponding key, until you have finally found the to-be-fetched data. (Can be empty/null). If the data is already present, just continue by calling `onlyBranchIndex` until you either find a node which does not exist in the DB, find out the value, or find out the value is empty.

Reply:
I just realized I have to edit the type a bit since it seems that the `OnNode` always has a `Trie` node but this is not the case.